concent_api/concent_api/middleware.py

```diff
@@ -39,3 +43,33 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+class HandleServerErrorMiddleware(object):
+    """
+    Used to catch all unhandled exceptions and return JSON response containing error information.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.sid = None
+
+    def __call__(self, request):
+        self.sid = transaction.savepoint()
+        response = self.get_response(request)
+        transaction.savepoint_commit(self.sid)
+        return response
+
+    def process_exception(self, request, exception):  # pylint: disable=unused-argument
+        transaction.savepoint_rollback(self.sid)
+        logging.logger.exception(exception)
+        return self._build_json_response(exception)
+
+    def _build_json_response(self, exception):  # pylint: disable=no-self-use
+        message = {"error": str(exception)}
```

Comment:
Please include an `error_code` field and set it to `'crash'`.

Also, you might check if the exception has an `error_code` field and it so, use this code instead (@rwrzesien included it in some exception classes in #133 and I think we'll be adding it to the new ones too). Maybe even it would be a good idea to add a base exception class with this field as a part of your pull request.

Reply:
I meant the base exception class that has the `error_code` field. There was one in @rwrzesien's pull request for JSON error codes.